src/core/friendly_errors/sketch_verifier.js

```diff
@@ -0,0 +1,118 @@
+import * as espree from 'espree';
+
+/**
+ * @for p5
+ * @requires core
+ */
+function sketchVerifier(p5, fn) {
+  /**
+   * Fetches the contents of a script element in the user's sketch.
+   * 
+   * @method fetchScript
+   * @param {HTMLScriptElement} script
+   * @returns {Promise<string>}
+   */
+  fn.fetchScript = async function (script) {
+    if (script.src) {
+      const contents = await fetch(script.src).then((res) => res.text());
+      return contents;
+    } else {
+      return script.textContent;
+    }
+  }
+
+  /**
+   * Extracts the user's code from the script fetched. Note that this method
+   * assumes that the user's code is always the last script element in the
+   * sketch.
+   * 
+   * @method getUserCode
+   * @returns {Promise<string>} The user's code as a string.
+   */
+  fn.getUserCode = async function () {
+    const scripts = document.querySelectorAll('script');
+    const userCodeScript = scripts[scripts.length - 1];
```

Comment:
I wonder if there's a more robust way to get the user's script, since e.g. some sketches are spread across multiple files, or might import other addon libraries. We definitely don't need to be perfect here, but since multi-file sketches are pretty common on OpenProcessing, maybe we could do something like, filter out all the scripts that look like library code, e.g. because their `src` attribute matches `/p5(\.\w+)\.js/`, and then run the parser on the remaining scripts?

We could also exclude srcs that come from a known CDN domain like `cdnjs.cloudflare.com` or `cdn.jsdelivr.net` or `unpkg.com`. This is what OpenProcessing uses for its additional library toggles, so we could exclude more library code that way, although with some false negatives still.

Reply:
One way I was thinking about is to only grab from `<script>` tag that are in `<body>` which assumes that library import `<script>` tags are usually included in the `<head>` element while user code are included in the `<body>` tag, but that is mostly convention and not a guarantee so I'm not sure if it would be a good solution.

The other thing is that, if I remember correctly, if any `<script>` tag is using `type=module` they shouldn't be grabbed as well because these are scoped automatically.

Reply:
Added a TODO!